packages/training/python/tests/e2e/test_online_training_e2e.py

```
@@ -0,0 +1,365 @@
+            "headline": "BTC is rising",
+            "sentiment": "bullish",
+            "impact": "high",
+            "source": "CryptoNews",
+        })
+        
+        assert len(scenario.news) == 1
+        assert scenario.news[0].headline == "BTC is rising"
+    
+    def test_scenario_metadata(self):
+        """Test Scenario.metadata field for extensibility"""
+        from src.training.scenario_pool import Scenario, PortfolioState
```

Ruff (0.14.10)

Redefinition of unused `PortfolioState` from line 41

Remove definition: `PortfolioState`

(F811)